fe/fe-core/src/main/java/com/starrocks/external/iceberg/cost/IcebergTableStatisticCalculator.java

```diff
@@ -66,27 +66,24 @@ private List<ColumnStatistic> makeColumnStatistics(List<Expression> icebergPredi
         List<Types.NestedField> columns = icebergTable.schema().columns();
         IcebergFileStats icebergFileStats = new IcebergTableStatisticCalculator(icebergTable).
                 generateIcebergFileStats(icebergPredicates, columns);
-        if (icebergFileStats == null) {
-            columnStatistics.add(ColumnStatistic.unknown());
-        } else {
-            Map<Integer, String> idToColumnNames = columns.stream()
-                    .filter(column -> column.type().isPrimitiveType())
-                    .collect(Collectors.toMap(Types.NestedField::fieldId, column -> column.name()));
-
-            double recordCount = Math.max(icebergFileStats.getRecordCount(), 1);
-            for (Map.Entry<Integer, String> idColumn : idToColumnNames.entrySet()) {
-                List<ColumnRefOperator> columnList = colRefToColumnMetaMap.keySet().stream().filter(
-                        key -> key.getName().equalsIgnoreCase(idColumn.getValue())).collect(Collectors.toList());
-                if (columnList == null || columnList.size() != 1) {
-                    LOG.debug("This column is not required column name " + idColumn.getValue() + " column list size "
-                            + (columnList == null ? "null" : columnList.size()));
-                    continue;
-                }
 
-                int fieldId = idColumn.getKey();
-                columnStatistics.add(
-                        generateColumnStatistic(icebergFileStats, fieldId, recordCount, columnList.get(0)));
+        Map<Integer, String> idToColumnNames = columns.stream()
+                .filter(column -> column.type().isPrimitiveType())
+                .collect(Collectors.toMap(Types.NestedField::fieldId, column -> column.name()));
+
+        double recordCount = Math.max(icebergFileStats == null ? 0 : icebergFileStats.getRecordCount(), 1);
+        for (Map.Entry<Integer, String> idColumn : idToColumnNames.entrySet()) {
+            List<ColumnRefOperator> columnList = colRefToColumnMetaMap.keySet().stream().filter(
+                    key -> key.getName().equalsIgnoreCase(idColumn.getValue())).collect(Collectors.toList());
+            if (columnList == null || columnList.size() != 1) {
+                LOG.debug("This column is not required column name " + idColumn.getValue() + " column list size "
+                        + (columnList == null ? "null" : columnList.size()));
+                continue;
             }
+
+            int fieldId = idColumn.getKey();
+            columnStatistics.add(
```

Comment:
@hongli-my Cuz we are not able to get ndv from table file, so the column statistics will be set to unknown any way. cc @caneGuy @dirtysalt